### PR TITLE
Add Page SEO

### LINKED ISSUE #562: @/codeforafrica seo
Implement SEO

The implementation should also keep in mind https://github.com/CodeForAfrica/ui/pull/443 since this approach promises to minimise changes should content editors move blocks around.

Current content:
https://github.com/CodeForAfrica/ui/blob/daacc18ac4e896494b8c7ae9fe84a978b2c2afd8/apps/codeforafrica/src/lib/api.netlify-cms/seo.js

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description
This PR adds individual Page SEO. This PR does not handle SEO for nested pages. This will be handled by #584 

Fixes #562

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/lib/data/rest/index.js [hunk 1]
```diff
@@ -37,6 +37,6 @@ export const api = {
 export async function getPageStaticProps(context) {
   const props = await getPageProps(api, context);
   return {
-    props,
+    props: props || {},
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/rest/index.js:40]: Why are we forcing it here?

File: apps/codeforafrica/src/lib/data/seo.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import site from "@/codeforafrica/utils/site";
+
+function stringifyDescription(description) {
+  if (!description || !Array.isArray(description)) {
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/seo.js:4]: Isn't 👇🏽 enough? i.e. Array.isArray handles null/undefined and description will always be an array if present, no?

```suggestion
  if (!Array.isArray(description)) {
```

File: apps/codeforafrica/src/lib/data/seo.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    return "";
+  }
+  let result = "";
+  description.forEach((item) => {
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/seo.js:8]: Not a fan of forEach... I think map or reduce are always better.

File: apps/codeforafrica/src/lib/data/seo.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+export default function getPageSeoFromMeta(page, settings) {
+  const { title: pageTitle, meta: pageMeta } = page;
+  const {
+    title: metaTitle,
+    description: metaDescription,
+    image = {},
+  } = pageMeta;
+  const { title: siteTitle, description: siteDescription } = settings;
+  const title = metaTitle || pageTitle || siteTitle || null;
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/seo.js:28]: Don't we have site title also defined in env vars? APP_NAME or APP_TITLE?

**Review by koechkevin — APPROVED** [review]

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

There is a way we're handling alternative hrefs or urls in SEO and I don't see it here. Is it because we don't support localisation?

File: apps/codeforafrica/src/lib/data/rest/index.js (hunk 1 above)
Comment (kelvinkipruto) [reply apps/codeforafrica/src/lib/data/rest/index.js:40]: @kilemensi This is to handle an edge case where the error page has not been created in the CMS yet.

**Comment by kelvinkipruto** [comment]
> There is a way we're handling alternative hrefs or urls in SEO and I don't see it here. Is it because we don't support localisation?

@kilemensi, Yes. We do not support localisation yet.

**Comment by kelvinkipruto** [comment]
@kilemensi @koechkevin This is ready

**Comment by kilemensi** [comment]
Can we update this PR @kelvinkipruto ?

**Comment by kelvinkipruto** [comment]
@kilemensi updated

File: apps/codeforafrica/src/lib/data/seo.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    null;
+  const description =
+    metaDescription || stringifyDescription(siteDescription) || null;
+  const titleTemplate = siteTitle ? `%s | ${siteTitle}` : null;
+  const defaultTitle = siteTitle || null;
+  const canonical = site.url.replace(/\/+$/, "");
+  const openGraph = {
+    title,
+    description,
+    type: "website",
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/seo.js:43]: Add a story to track these: we may have to use different types for stories, teams, etc.

**Review by kilemensi — APPROVED** [review]

File: apps/codeforafrica/src/lib/data/seo.js (hunk 5 above)
Comment (kelvinkipruto) [reply apps/codeforafrica/src/lib/data/seo.js:43]: @kilemensi Already there at #584

**Pull request merged**